sklearn/dummy.py

```diff
@@ -122,7 +123,11 @@ def fit(self, X, y, sample_weight=None):
         if not self.sparse_output_:
             y = np.atleast_1d(y)
 
-        self.output_2d_ = y.ndim == 2
+        self.output_2d_ = (y.ndim == 2) and (y.shape[1] > 1)
+
+        if not self.output_2d_:
+            y = column_or_1d(y, warn=True)
```

Comment:
This is reversed right after when `y` is reshaped to `(-1, 1)`. At least as far as the added tests here go, This if and `column_or_1d` is not needed and the change for `output_2d_` fixes the issue. Shouldn't the same fix apply to dummyregressor?